fees: enable `CBlockPolicyEstimator` return sub 1 sat/vb fee rate estimates

ismaelsadeeq opened the pull request:
This is a simple PR that updates the block policy estimator’s `MIN_BUCKET_FEERATE` constant to be 100, which is identical to the policy `DEFAULT_MIN_RELAY_TX_FEE`. 

This change enables the block policy fee rate estimator to return sub-1 sat/vB fee rate estimates.

The change is correct because the estimator creates buckets of fee rates from
`MIN_BUCKET_FEERATE`,
`MIN_BUCKET_FEERATE` + `FEE_SPACING`,
`MIN_BUCKET_FEERATE` + `2 * FEE_SPACING`,
… up to `MAX_BUCKET_FEERATE`.

This means it will record sub-1 sat/vB fee rates in the buckets and may return them as a fee rate estimate when that bucket is the lowest one with sufficient transactions for a given target.

---

While touching this part of the fee estimator code, this PR got rid of the dummy value persisted in the file

polespinasa reviewed (commented):
concept ACK

This change makes much sense to me, if the min relay fee is lowered the feerate tracking should also take that into account.

kevkevinpal commented on test/functional/feature_fee_estimation.py:
```diff
@@ -412,20 +417,24 @@ def broadcast_and_mine(self, broadcaster, miner, feerate, count):
         # Check the fee estimates for both modes after mining low fee transactions.
         check_fee_estimates_btw_modes(self.nodes[0], high_feerate, low_feerate)
 
+    def test_sub_1s_per_vb_estimates(self):
```
Just an fyi, this test passes even without the changes in `src/policy/fees.h` I also agree with @polespinasa that if another test already covers this, then I'd rather not have redundant tests added

kevkevinpal commented on src/policy/fees.h:
```diff
@@ -180,13 +181,10 @@ class CBlockPolicyEstimator : public CValidationInterface
     /** Minimum and Maximum values for tracking feerates
-     * The MIN_BUCKET_FEERATE should just be set to the lowest reasonable feerate we
-     * might ever want to track.  Historically this has been 1000 since it was
-     * inheriting DEFAULT_MIN_RELAY_TX_FEE and changing it is disruptive as it
-     * invalidates old estimates files. So leave it at 1000 unless it becomes
```
I am not sure if we need a release note, since according to this note, it will invalidate old estimates files? Maybe someone else can chime in on this

ismaelsadeeq replied:
Done

ismaelsadeeq committed:
fees: reduce `MIN_BUCKET_FEERATE` to 100

- The DEFAULT_MIN_RELAY_TX_FEE has been reduced to
  100 in v30. This change updates block policy fee estimator
  to reflect that change.

ismaelsadeeq committed:
test: ensure fee estimator provide fee rate estimate < 1 s/vb

- This commit also refactors the test the better track confirmed
  utxos.

ismaelsadeeq committed:
doc: add release notes

polespinasa commented on src/policy/fees/block_policy_estimator.cpp:
```diff
@@ -33,8 +33,8 @@
 // The current format written, and the version required to read. Must be
-// increased to at least 289900+1 on the next breaking change.
-constexpr int CURRENT_FEES_FILE_VERSION{149900};
+// increased to at least 309900+1 on the next breaking change.
```
Commit a1a1bfa2445da11369323b649253f81d288f2813  feels out of scope for this PR.

just curiosity, how is this number chosen? Related to release number?

polespinasa commented on src/policy/fees/block_policy_estimator.cpp:
```diff
@@ -980,7 +980,6 @@ bool CBlockPolicyEstimator::Write(AutoFile& fileout) const
     try {
         LOCK(m_cs_fee_estimator);
         fileout << CURRENT_FEES_FILE_VERSION;
-        fileout << int{0}; // Unused dummy field. Written files may contain any value in [0, 289900]
```
this changes feel out of scope for this pr

polespinasa reviewed (commented):
Just some nits and questions :=)

ismaelsadeeq commented on src/policy/fees/block_policy_estimator.h:
```diff
@@ -181,13 +182,10 @@ class CBlockPolicyEstimator : public CValidationInterface
-     * inheriting DEFAULT_MIN_RELAY_TX_FEE and changing it is disruptive as it
-     * invalidates old estimates files. So leave it at 1000 unless it becomes
-     * necessary to lower it, and then lower it substantially.
+     * The MIN_BUCKET_FEERATE should just be set to the lowest reasonable feerate
```
What you mentioned is exactly what the comment is now.

ismaelsadeeq committed:
fees: bump fees file version

ismaelsadeeq committed:
fees: delete unused dummy field

- This commit deletes a dummy field that is
  "dummy" and unused.

polespinasa reviewed (commented):
code looks good to me, currently trying to test it on mainnet will ack if positive results :)

a small nit for more clarity on the PR

polespinasa reviewed (approved):
cr and tACK b76789787c43802aaf1207813792ea49ecf18f55

achow101 commented:
ACK b76789787c43802aaf1207813792ea49ecf18f55

maflcko reviewed (commented):
Left some tiny nits. Feel free to ignore/close/dismiss.

ismaelsadeeq commented:
Forced pushed from bb0bf4ac18024521f7de6969a501e03198c44982 to 8966352df3fc56fd2c00a45eecd292a240a34546 compare diff [8c44982..89663](https://github.com/bitcoin/bitcoin/compare/bb0bf4ac18024521f7de6969a501e03198c44982..8966352df3fc56fd2c00a45eecd292a240a34546)

I Addressed LLM comment and nit from @maflcko

musaHaruna reviewed (commented):
ACK [8966352](https://github.com/bitcoin/bitcoin/commit/8966352df3fc56fd2c00a45eecd292a240a34546)

polespinasa reviewed (approved):
ACK 8966352df3fc56fd2c00a45eecd292a240a34546
